src/index.jsx

```diff
@@ -2,9 +2,12 @@ import React from 'react'
 import {createRoot} from 'react-dom/client'
 import {BrowserRouter} from 'react-router-dom'
 import BaseRoutes from './BaseRoutes'
-
+import WidgetApi from "./WidgetApi/WidgetApi";
 
 const root = createRoot(document.getElementById('root'))
+
+new WidgetApi()
```

Comment:
The Problem here is that we need to initialize this class at a place which doesn't get reinitialized. I debugged this and the outcome was that any component gets reloaded at routing time. Since both ends of matrix-widget-api relies on a specific order of events there is no stable communication when initialized in routed components.

Reply:
but i'm not specialized in building react-apps maybe you guys know a better way of doing this :)

Reply:
I don't know the answer, but I think this is worth solving.  Happy to help debug together.  Have you tried in src/Share.jsx or its parent components?  It only rerenders on new models, but I think that's probably desirable here.

Reply:
i tested it again today and it only works stable at this place